source/adapters/level_zero/command_buffer.cpp

```diff
@@ -986,3 +986,93 @@ UR_APIEXPORT ur_result_t UR_APICALL urCommandBufferEnqueueExp(
 
   return UR_RESULT_SUCCESS;
 }
+
+UR_APIEXPORT ur_result_t UR_APICALL urEventGetSyncPointProfilingInfoExp(
+    ur_event_handle_t Event, ur_exp_command_buffer_sync_point_t SyncPoint,
+    ur_profiling_info_t PropName, size_t PropValueSize, void *PropValue,
+    size_t *PropValueSizeRet) {
+  std::shared_lock<ur_shared_mutex> EventLock(Event->Mutex);
+
+  if (Event->UrQueue &&
+      (Event->UrQueue->Properties & UR_QUEUE_FLAG_PROFILING_ENABLE) == 0) {
+    return UR_RESULT_ERROR_PROFILING_INFO_NOT_AVAILABLE;
+  }
+
+  ur_device_handle_t Device =
+      Event->UrQueue ? Event->UrQueue->Device : Event->Context->Devices[0];
+
+  uint64_t ZeTimerResolution = Device->ZeDeviceProperties->timerResolution;
+  const uint64_t TimestampMaxValue =
+      ((1ULL << Device->ZeDeviceProperties->kernelTimestampValidBits) - 1ULL);
+
+  UrReturnHelper ReturnValue(PropValueSize, PropValue, PropValueSizeRet);
+
+  // Node profiling info is stored in the CommandData field of the event
+  // returned from graph submission.
+  // The timing info of each command corresponding to a node is stored using
+  // `zeCommandListAppendQueryKernelTimestamps`. This command stores the timing
+  // info of each command/node in the same order as the sync-points that were
+  // assigned to this node when it was enqueued. Consequently, `SyncPoint`
+  // corresponds to the index of the memory slot containing the timestamps
+  // corresponding to this specific node.
+  if (Event->CommandType == UR_COMMAND_COMMAND_BUFFER_ENQUEUE_EXP) {
+    if (Event->CommandData) {
```

Comment:
Would prefer to make these early returns and remove some indentation from the interesting part of the code.
```cpp
if (Event->CommandType != UR_COMMAND_COMMAND_BUFFER_ENQUEUE_EXP) {
   urPrint("urEventGetSyncPointProfilingInfoExp: not supported Event type\n");
    return UR_RESULT_ERROR_INVALID_VALUE;
}

if (!Event->CommandData) {
  return UR_RESULT_ERROR_PROFILING_INFO_NOT_AVAILABLE;
}
```

Reply:
The code has been updated this way